### DISCUSSION TITLE
ROS2 command in terminal.

### DISCUSSION THREAD

**Discussion by YunwooLee94** [microsoft/AirSim#4833] [Support Q&A] [answered]
I am trying to fly a drone using ROS2 wrapper.
Could you share an example of using "ros2 topic pub" on the terminal to publish "airsim_node/<drone_name>/vel_cmd_body_frame" in order to control a drone using ROS2 wrapper?

ros2 topic pub -r 10 /airsim_node/Drone1/vel_cmd_world_frame airsim_interfaces/msg/VelCmd twist:\
\ \ linear:\
\ \ \ \ x:\ 0.0\
\ \ \ \ y:\ 0.0\
\ \ \ \ z:\ 1.0\
\ \ angular:\
\ \ \ \ x:\ 0.0\
\ \ \ \ y:\ 0.0\
\ \ \ \ z:\ 0.0\ 

This causes following errors.

yaml.scanner.ScannerError: mapping values are not allowed here
  in "<unicode string>", line 1, column 15:
    twist:  linear:    x: 0.0    y: 0.0    z: 1.0  ... 

In addition, I use ros-foxy.

Thanks all

**Comment by YunwooLee94** [comment, marked as answer]
Instead of using "tab", I used "{twist: {linear:{x: 0.0, y: 0.0, z: -1.0}, angular:{x: 0.0, y: 0.0, z: 0.0}}}".
And it worked.

**Comment by EXPX3** [comment]
@YunwooLee94 
**I am trying to run a basic example of roswrapper and airsim. drone takeoff!**
**Below is my setup.**
- airsim and UR in docker container by runing the command ./run_airsim_image_binary.sh airsim_binary:11.3.1-devel-ubuntu18.04 Blocks/LinuxNoEditor/Blocks.sh -windowed -ResX=1080 -ResY=720'
- roswrapper in docker container by running the command docker run --rm -it --network=test_network --name ros2wrapper -v <path to airsim in host>:/home/testuser/AirSim airsim-ros2:latest bash
- until this point i am able to run both docker containers and start blocks environment via airsim related container and run airsim_node via roswrapper vontainer, i have made sure these both docker containers communicate via a test_network i created. i can ping from one container to another.

**The Problem !**
when i run ros2 service list i dont see the takeoff, land and other publishers and subscribers listed in the doccumentation page [here](https://microsoft.github.io/AirSim/airsim_ros_pkgs/)

Can you please tell me if i am missing any crusial step inbetween? and how to proceed from here?

Many Thanky in advance!!

**Reply by EXPX3** [reply]
Hi All, when i looked at the output of the launching the airsim_node via the launch file, i noticed that the airsim_node is named as [airsim_node-2], this number means the number of similar nodes that are running. Ideally we need only one airsim_node running. so instead of running the launch file i just launched the aursim_node using ros2 run airsim_ros_pkgs airsim_node command, and the out put is
waiting for the connection xxxx

THis means that the roswrapper is not connected to the airsim simulation environment or?
how should i make sure that they are connected? how did you guys manage to set it up?, in the doccumentation i did not come across any extra step to connect the roswrapper docker and airsim from binary docker.
Any pointers or help is welcome please!

Thanks in advance.
